tox.ini

```diff
@@ -1,42 +1,34 @@
 [tox]
 toxworkdir={env:TOXWORKDIR:{toxinidir}/.tox}
 envlist =
-    {py35,py36,py37}-django20
-    {py35,py36,py37}-django21
-    {py35,py36,py37,py38,py39}-django22
-    {py36,py37,py38,py39}-django30
-    {py36,py37,py38,py39}-django31
-    {py36,py37,py38,py39,py310}-django32
+    {py38,py39,py310}-django32
     {py38,py39,py310}-django40
-    {py38,py39,py310}-django41
-    {py38,py39,py310}-django42
+    {py38,py39,py310,py311}-django41
+    {py38,py39,py310,py311}-django42
+    {py310,py311,py312}-django42
     flake8
 skip_missing_interpreters = True
 
 [flake8]
 max-line-length = 120
 
 [testenv]
-commands = python ./manage.py test
+commands = python -W error::DeprecationWarning -W error::PendingDeprecationWarning ./manage.py test
 envdir = {toxworkdir}/venvs/{envname}
 setenv =
     PYTHONDONTWRITEBYTECODE=1
     PYTHONWARNINGS=once
     DJANGO_SETTINGS_MODULE=tests.settings
 deps =
-    django20: django>=2.0,<2.1
-    django21: django>=2.1,<2.2
-    django22: django>=2.2,<3.0
-    django30: django>=3.0,<3.1
-    django31: django>=3.1,<3.2
     django32: django>=3.2,<3.3
     django40: django>=4.0,<4.1
     django41: django>=4.1,<4.2
     django42: django>=4.2,<4.3
+    django50: django>=5.0a1,<5.1
 
 [testenv:flake8]
-basepython = python3.7
+basepython = python3.12
 deps =
-    flake8==3.8.4
+    flake8==6.1.0
```

Comment:
Latest Flake8: mostly performance improvements and compatibility fixes. I don't think any of the errors I see are from new rules:

```
./django_lifecycle/dataclass_validation.py:31:26: F541 f-string is missing placeholders
./django_lifecycle/mixins.py:245:5: E303 too many blank lines (2)
./pypi_submit.py:4:33: W292 no newline at end of file
./tests/testapp/tests/test_mixin.py:103:121: E501 line too long (121 > 120 characters)
./tests/testapp/tests/test_mixin.py:104:30: E128 continuation line under-indented for visual indent
./tests/testapp/tests/test_mixin.py:110:121: E501 line too long (122 > 120 characters)
./tests/testapp/tests/test_mixin.py:111:30: E128 continuation line under-indented for visual indent
./tests/testapp/tests/test_mixin.py:128:30: E128 continuation line under-indented for visual indent
./tests/testapp/tests/test_mixin.py:128:121: E501 line too long (146 > 120 characters)
./tests/testapp/tests/test_mixin.py:134:121: E501 line too long (122 > 120 characters)
./tests/testapp/tests/test_mixin.py:135:30: E128 continuation line under-indented for visual indent
./tests/testapp/tests/test_mixin.py:350:25: E251 unexpected spaces around keyword / parameter equals
./tests/testapp/tests/test_mixin.py:350:27: E251 unexpected spaces around keyword / parameter equals
./tests/testapp/tests/test_mixin.py:354:121: E501 line too long (126 > 120 characters)
./tests/testapp/tests/test_mixin.py:355:30: E128 continuation line under-indented for visual indent
./tests/testapp/tests/test_mixin.py:361:121: E501 line too long (126 > 120 characters)
./tests/testapp/tests/test_mixin.py:362:30: E128 continuation line under-indented for visual indent
./tests/testapp/tests/test_mixin.py:368:121: E501 line too long (126 > 120 characters)
./tests/testapp/tests/test_mixin.py:369:30: E128 continuation line under-indented for visual indent
./tests/testapp/tests/test_mixin.py:375:121: E501 line too long (124 > 120 characters)
./tests/testapp/tests/test_mixin.py:376:30: E128 continuation line under-indented for visual indent
./tests/testapp/tests/test_mixin.py:387:121: E501 line too long (150 > 120 characters)
./tests/testapp/tests/test_user_account.py:130:121: E501 line too long (129 > 120 characters)
```